x-pack/plugins/alerting/server/task_runner/lib/clear_expired_snoozes.ts

```diff
@@ -5,44 +5,41 @@
  * 2.0.
  */
 
-import { RuleTypeParams, SanitizedRule } from '../../../common';
-import { partiallyUpdateRule } from '../../saved_objects';
+import { RuleTypeParams, SanitizedRule } from '@kbn/alerting-types';
+import { ISavedObjectsRepository, Logger } from '@kbn/core/server';
 import { isSnoozeExpired } from '../../lib';
-import { RulesClientContext } from '../types';
-import { updateMeta } from '../lib';
-
-export async function clearExpiredSnoozes(
-  context: RulesClientContext,
-  {
-    rule,
-    version,
-  }: { rule: Pick<SanitizedRule<RuleTypeParams>, 'id' | 'snoozeSchedule'>; version?: string }
-): Promise<void> {
+import { partiallyUpdateRule } from '../../saved_objects';
+
+interface ClearExpiredSnoozesOpts {
+  logger: Logger;
+  rule: Pick<SanitizedRule<RuleTypeParams>, 'id' | 'snoozeSchedule'>;
+  savedObjectsClient: ISavedObjectsRepository;
+  version?: string;
+}
+export async function clearExpiredSnoozes(opts: ClearExpiredSnoozesOpts): Promise<void> {
+  const { logger, rule, savedObjectsClient, version } = opts;
+
+  if (!rule.snoozeSchedule || !rule.snoozeSchedule.length) return;
+
   const snoozeSchedule = rule.snoozeSchedule
     ? rule.snoozeSchedule.filter((s) => {
         try {
           return !isSnoozeExpired(s);
         } catch (e) {
-          context.logger.error(`Error checking for expiration of snooze ${s.id}: ${e}`);
+          logger.error(`Error checking for expiration of snooze ${s.id}: ${e}`);
           return true;
         }
       })
     : [];
 
   if (snoozeSchedule.length === rule.snoozeSchedule?.length) return;
 
-  const updateAttributes = updateMeta(context, {
+  const updateAttributes = {
     snoozeSchedule,
-    updatedBy: await context.getUserName(),
```

Comment:
Since it's the alerting framework updating the saved object, we should not need to identify a username for this update, we don't do it when updating the rule SO after each rule execution.

Reply:
Also removed `updatedAt` since we don't update that value when we make other partial updates to the rule during rule execution and this limits the number of fields we allow to update via ES.